### PR TITLE
Minor fixes in publish test results

### PR DESCRIPTION
**Pull request by AshwiniChalla**
Issue: when there are no matching files matchingTestResultsFiles is an empty array. 
Fix: verify the length of the array before calling command.

### CODE REVIEW

**Comment by acesiddhu** [comment]
looks fine

**Comment by kaadhina** [comment]
Looks fine

File: Tasks/PublishTestResults/publishtestresults.ts [hunk 1]
```diff
@@ -30,10 +30,11 @@ else {
...
-if(!matchingTestResultsFiles) {
+if(!matchingTestResultsFiles || matchingTestResultsFiles.length == 0) {
   tl.warning('No test result files matching ' + testResultsFiles + ' were found.');  
   tl.exit(0);
 }
-
-var tp = new tl.TestPublisher(testRunner);
-tp.publish(matchingTestResultsFiles, mergeResults, platform, config, testRunTitle, publishRunAttachments);
\ No newline at end of file
+else{
```
Comment (SatishThareja) [inline Tasks/PublishTestResults/publishtestresults.ts]: won't 'tl.exit', exit the script? In that case code can be taken out of else block.

File: Tasks/PublishTestResults/publishtestresults.ts [hunk 2]
```diff
@@ -30,10 +30,11 @@ else {
   var matchingTestResultsFiles = [testResultsFiles];
 }
 
-if(!matchingTestResultsFiles) {
+if(!matchingTestResultsFiles || matchingTestResultsFiles.length == 0) {
```
Comment (prawalagarwal) [inline Tasks/PublishTestResults/publishtestresults.ts]: use === where ever possible

File: Tests/lib/vsts-task-lib/task.ts [hunk 3]
```diff
@@ -608,9 +608,19 @@ export class TestPublisher {
             properties['config'] = config;
         }
 
-        for (var i = 0; i < resultFiles.length; i++) {
-            command('results.publish', properties, resultFiles[i]);
+        if (runTitle) {
+            properties['runTitle'] = runTitle;
```
Comment (prawalagarwal) [inline Tests/lib/vsts-task-lib/task.ts]: In case no run title is provided, is the behavior same as before (for run titles)?

Comment (AshwiniChalla) [inline Tests/lib/vsts-task-lib/task.ts]: Yes old behavior if runtitle is not passed

File: Tasks/PublishTestResults/publishtestresults.ts (hunk 2 above)
Comment (AshwiniChalla) [inline Tasks/PublishTestResults/publishtestresults.ts]: I think == is strongly typed and === is not strongly typed. I guess it is better to use == in typescript as it complies with language rules. Let me know your thoughts

File: Tasks/PublishTestResults/publishtestresults.ts (hunk 1 above)
Comment (AshwiniChalla) [inline Tasks/PublishTestResults/publishtestresults.ts]: tl.exit is not exiting . It just set the result. Thats why I moved it into else block.

Comment (bryanmacfarlane) [inline Tasks/PublishTestResults/publishtestresults.ts]: Note that tl.exit is deprecated but supported for back compat.  Abrubtly exiting the script (process.exit) can cause some output to be lost.  I will remove in next majoor update of task lib.  Just let the task exit and use tl.setresult to success or failure.

File: Tasks/PublishTestResults/publishtestresults.ts (hunk 2 above)
Comment (prawalagarwal) [inline Tasks/PublishTestResults/publishtestresults.ts]: [http://stackoverflow.com/questions/359494/does-it-matter-which-equals-operator-vs-i-use-in-javascript-comparisons](url)

**Pull request merged**